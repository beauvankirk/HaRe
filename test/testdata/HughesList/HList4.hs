module Main where

exponents :: Int -> [Int]
exponents base = base : (exponents (2*base))

main = putStrLn "hello"
