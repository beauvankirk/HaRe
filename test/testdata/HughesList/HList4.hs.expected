module HList4 where
import qualified Data.DList as DList
import Data.DList (DList)

exponents :: Int -> DList Int
exponents base = base `DList.cons` (exponents (2*base))
